From e87532dc34dbf78a47a0b9783dca2f69189c03a8 Mon Sep 17 00:00:00 2001
From: jenisys <jenisys@users.noreply.github.com>
Date: Sun, 9 Feb 2020 20:30:53 +0100
Subject: [PATCH] Clarify select-by-name uses regex pattern (related to: issue
 #810)

Clarifiy select-by-name uses regex pattern:

* Adapt command-line option --name help text
* Update command-line args docs for behave
---
 CHANGES.rst             |  4 ++++
 behave/configuration.py | 10 +++++-----
 docs/behave.rst         | 12 ++++++------
 3 files changed, 15 insertions(+), 11 deletions(-)

diff --git a/CHANGES.rst b/CHANGES.rst
index 5653492..d0bf6fd 100644
--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -31,6 +31,10 @@ ENHANCEMENTS:
 * issue #678: Scenario Outline: Support tags with commas and semicolons (provided by: lawnmowerlatte, pull #679)
 * issue #675: Feature files cannot be found within symlink directories (provided by: smadness, pull #680)
 
+CLARIFICATION:
+
+* issue #810: Clarify select-by-name using regex pattern (submitted by: xv-chris-w)
+
 FIXED:
 
 * issue #772: ScenarioOutline.Examples without table (submitted by: The-QA-Geek)
diff --git a/behave/configuration.py b/behave/configuration.py
index bd8b039..65e2e3e 100644
--- a/behave/configuration.py
+++ b/behave/configuration.py
@@ -164,11 +164,11 @@ options = [
                   override a configuration file setting.""")),
 
     (("-n", "--name"),
-     dict(action="append",
-          help="""Only execute the feature elements which match part
-                  of the given name. If this option is given more
-                  than once, it will match against all the given
-                  names.""")),
+     dict(action="append", metavar="NAME_PATTERN",
+          help="""Select feature elements (scenarios, ...) to run
+                  which match part of the given name (regex pattern).
+                  If this option is given more than once,
+                  it will match against all the given names.""")),
 
     (("--no-capture",),
      dict(action="store_false", dest="stdout_capture",
diff --git a/docs/behave.rst b/docs/behave.rst
index dfb390a..25ce523 100644
--- a/docs/behave.rst
+++ b/docs/behave.rst
@@ -95,9 +95,9 @@ You may see the same information presented below at any time using ``behave
 
 .. option:: -n, --name
 
-    Only execute the feature elements which match part of the given name.
-    If this option is given more than once, it will match against all
-    the given names.
+    Select feature elements (scenarios, ...) to run which match part of
+    the given name (regex pattern). If this option is given more than
+    once, it will match against all the given names.
 
 .. option:: --no-capture
 
@@ -449,9 +449,9 @@ Configuration Parameters
 
 .. describe:: name : sequence<text>
 
-    Only execute the feature elements which match part of the given name.
-    If this option is given more than once, it will match against all
-    the given names.
+    Select feature elements (scenarios, ...) to run which match part of
+    the given name (regex pattern). If this option is given more than
+    once, it will match against all the given names.
 
 .. index::
     single: configuration param; stdout_capture
